### ISSUE TITLE
scala_library doesn't support reference.conf

### ISSUE THREAD

**Issue by bjchambers** [higherkindness/rules_scala#216]
See this draft PR for an example demonstrating the issue:
https://github.com/higherkindness/rules_scala/pull/215

```
java_library(
    name = "plain_java",
    resources = glob(["reference.conf"]),
    resource_strip_prefix = "resources/reference_conf",
)
```

Produces a JAR containing the manifest and a single file `reference.conf`.

```
scala_library(
    name = "plain",
    resources = glob(["reference.conf"]),
    resource_strip_prefix = "resources/reference_conf",
    scala = "//scala:2_11",
)
```

Produces a JAR containing only the manifest.

It seems like the two libraries should both contain the `reference.conf` file.

**Comment by SrodriguezO** [comment]
@bjchambers thanks for filing this issue and for including a test that catches it. This does seem like a bug. I'll see if we can get a fix in over the next couple days.

**Comment by bjchambers** [comment]
It looks like this is an issue with the C++ implementation of singlejar. If I'm reading this right:

```
INFO: From singlejar resources plain:
singlejar_local: src/tools/singlejar/output_jar.cc:923: Duplicate resource name reference.conf in the --classpath_resource or --resource option
```

If I look at 
https://github.com/bazelbuild/bazel/blob/master/src/tools/singlejar/output_jar.cc#L933
I see this, which makes it look the resource is skipped if it is already present and warn_duplicate_resources is on:

```
  if (known_members_.count(resource_name)) {
    if (options_->warn_duplicate_resources) {
      diag_warnx(
          "%s:%d: Duplicate resource name %s in the --classpath_resource or "
          "--resource option",
          __FILE__, __LINE__, resource_name.c_str());
      // TODO(asmundak): this mimics old behaviour. Confirm that unless
      // we run with --warn_duplicate_resources, the output zip file contains
      // the concatenated contents of the all the resources with the same name.
      return;
    }
  }
```

Later in the same file, I see:

```
void OutputJar::ExtraCombiner(const std::string &entry_name,
                              Combiner *combiner) {
  extra_combiners_.emplace_back(combiner);
  known_members_.emplace(entry_name, EntryInfo{combiner});
}
```

And in the main file
https://github.com/bazelbuild/bazel/blob/master/src/tools/singlejar/singlejar_main.cc
I see:

```
  output_jar.ExtraCombiner("reference.conf",
                           new Concatenator("reference.conf"));
```

It doesn't seem like the Java implementation of singlejar has the same problems.

**Comment by SrodriguezO** [comment]
Very interesting. Good find! I'm surprised that Bazel skips the resource altogether if `warn_duplicate_resources` is specified.. 

I think the solution here is, as you saw, to stop passing that flag. If you'd like to promote your draft PR to a regular PR I will gladly approve it :)

**Comment by bjchambers** [comment]
Done -- Thanks!

**Comment by SrodriguezO** [comment]
Closed by #215
